sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,236 @@
+   * @param casSuppliers pending CAS suppliers.
+   * @param summary log summary.
+   * @param auth CAS auth details.
+   * @returns a number of update records.
+   */
+  private async requestCASAndUpdateSuppliers(
+    casSuppliers: CASSupplier[],
+    parentProcessSummary: ProcessSummary,
+    auth: CASAuthDetails,
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    for (const casSupplier of casSuppliers) {
+      const summary = new ProcessSummary();
+      parentProcessSummary.children(summary);
+      summary.info(`Requesting info for CAS supplier id ${casSupplier.id}.`);
+      let supplierResponse = null;
+      try {
+        supplierResponse = await this.casService.getSupplierInfoFromCAS(
+          auth.access_token,
+          casSupplier.student.sinValidation.sin,
+          casSupplier.student.user.lastName.toUpperCase(),
+        );
+        suppliersUpdated = await this.updateActiveAddress(
+          supplierResponse,
+          summary,
+          casSupplier,
+          suppliersUpdated,
+        );
+      } catch (error: unknown) {
+        summary.error(
+          "Error while requesting or updating CAS suppliers.",
+          error,
+        );
+      }
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * Updates supplier if finds an item from the response with an active address.
+   * @param supplierResponse CAS supplier response.
+   * @param summary log summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @param suppliersUpdated
+   * @returns a number of update records.
+   */
+  private async updateActiveAddress(
+    supplierResponse: CASSupplierResponse,
+    summary: ProcessSummary,
+    casSupplier: CASSupplier,
+    suppliersUpdated: number,
+  ) {
+    if (supplierResponse.items.length) {
+      const [supplierInfo] = supplierResponse.items;
+      const isThereAnActiveSupplierAddress =
+        this.getActiveSupplier(supplierInfo);
+      if (isThereAnActiveSupplierAddress) {
+        summary.info("Updating CAS supplier table.");
+        try {
+          const updateResult = await this.updateCASSupplier(
+            casSupplier.id,
+            supplierInfo,
+            SupplierStatus.Verified,
+          );
+          if (updateResult.affected) {
+            suppliersUpdated++;
+          }
+        } catch (error: unknown) {
+          throw new Error(
+            "Unexpected error while updating CAS supplier table.",
+            error,
+          );
+        }
+      } else {
+        summary.info("No active supplier address found on CAS.");
+      }
+    } else {
+      summary.info("No supplier found on CAS.");
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * Gets the first active supplier address from a list supplier response item.
+   * @param casSupplierResponseItem CAS supplier response item.
+   * @returns CAS supplier response item address.
+   */
+  private getActiveSupplier(
```

Comment:
The active status exists at two levels, the supplier itself and his addresses.
Should this method check for both?